modules/oe_whitelabel_search/src/Plugin/Block/SearchBlock.php

```diff
@@ -138,13 +138,7 @@ public function blockForm($form, FormStateInterface $form_state): array {
       '#tree' => TRUE,
       '#description' => $this->t('Fill in the settings of the Button field.'),
     ];
-    $form['button']['button_label'] = [
```

Comment:
Better put label back and add the config to decide upon three cases: icon, label and icon+label. Regarding this, move the value at the pattern above.